packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -382,6 +450,40 @@ export class Pipeline extends cdk.Construct implements cpapi.IPipeline {
   }
 
   private renderStages(): cloudformation.PipelineResource.StageDeclarationProperty[] {
+    // handle cross-region CodePipeline overrides here
+    let crossRegion = false;
+    this.stages.forEach((stage, i) => {
+      stage.actions.forEach((action, j) => {
+        if (action.region) {
+          crossRegion = true;
+          this.pipelineResource.addPropertyOverride(`Stages.${i}.Actions.${j}.Region`, action.region);
+        }
+      });
+    });
+
+    if (crossRegion) {
+      // we don't need ArtifactStore in this case
+      this.pipelineResource.addPropertyDeletionOverride('ArtifactStore');
```

Comment:
Try to avoid using property overrides altogether in favor of lazy rendering (Tokens)

Reply:
We have to use property overrides here, as currently the L1 validates `artifactStore` as a required field.